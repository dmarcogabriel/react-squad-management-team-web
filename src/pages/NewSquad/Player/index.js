import React from 'react';
import { MdAdd } from 'react-icons/md';
import cn from 'classnames';
import { useDrop } from 'react-dnd';
import { showPlayerInitials } from '../../../utils/playersInitials';

const Player = ({ className, player, onDropPlayer }) => {
  const [{ isOver }, drop] = useDrop({
    accept: 'player',
    drop: (e) => onDropPlayer(e.id),
    collect: (monitor) => ({
      isOver: !!monitor.isOver(),
    }),
  });

  const hasPlayer = () =>
    player.player
      ? 'bg-opacity-100 text-primary-dark'
      : 'bg-opacity-50 text-white border-2';

  return (
    <div
      data-testid="drop-player"
      ref={drop}
      className={cn(
        'relative rounded-full border-dashed p-8',
        'bg-white inline-block text-white',
        isOver && 'bg-opacity-100',
        hasPlayer(),
        className
      )}
    >
      <div
        className={cn(
          'absolute top-0 left-0',
          'flex items-center justify-center w-full h-full'
        )}
      >
        {player.player ? (
          <p className="text-base">
            {showPlayerInitials(player.player.display_name)}
          </p>
        ) : (
          <MdAdd size={22} />
        )}
      </div>
    </div>
  );
};

export default Player;
