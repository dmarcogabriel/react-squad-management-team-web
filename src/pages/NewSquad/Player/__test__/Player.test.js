import React from 'react';
import ReactDOM from 'react-dom';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Player from '..';

const mockNullPlayer = {};
const mockPlayer = {
  player: {
    display_name: 'Mock Name',
  },
};

const Component = (props) => (
  <DndProvider backend={HTML5Backend}>
    <Player {...props} />
  </DndProvider>
);

describe('<Player />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Component player={mockNullPlayer} />, div);
  });

  it('render classes correctly when player is not null', () => {
    const { getByTestId } = render(<Component player={mockPlayer} />);

    const dropZone = getByTestId('drop-player');

    expect(dropZone.className).toEqual(
      'relative rounded-full border-dashed p-8 bg-white inline-block text-white bg-opacity-100 text-primary-dark'
    );
  });

  it('match snapshot', () => {
    expect(shallow(<Component player={mockNullPlayer} />)).toMatchSnapshot();
  });
});
