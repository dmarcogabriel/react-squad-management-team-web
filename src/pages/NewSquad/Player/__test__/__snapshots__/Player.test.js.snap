// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> match snapshot 1`] = `
<DndProvider
  backend={[Function]}
>
  <Player
    player={Object {}}
  />
</DndProvider>
`;
